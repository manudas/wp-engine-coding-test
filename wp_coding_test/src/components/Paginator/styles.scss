@import '../globals';

.paginator {
    &__page {
        margin: 0 5px;
    }

    &__button {
        transition: all 0.3s ease-out;

        border-radius: 3px;
        border: 1px solid $semiTransparentBlack;

        &:hover {
            cursor: pointer;
            background-color: $semiTransparentWhite;
            color: $darkerOrange;
            border: 1px solid $grey;
        }

        &.disabled {
            pointer-events: none;
            opacity: 0.7;
        }
    }
}
