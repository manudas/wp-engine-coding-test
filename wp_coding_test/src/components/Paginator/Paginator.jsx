import React from 'react';

import PropTypes from 'prop-types';

import {
    buttonClass,
} from '../constants';

import './styles.scss';

const Paginator = ({
    page,
    setPage,
    maxPage,
}) => {

    return (
        <nav>
            <button
                className={`${buttonClass} ${page === 0 ? 'disabled' : ''}`}
                onClick={() => setPage(page - 1)}
            >
                Previous
            </button>
            <span
                className="paginator__page"
            >
                Page {page + 1} out of {maxPage + 1}
            </span>
            <button
                className={`${buttonClass} ${page === maxPage ? 'disabled' : ''}`}
                onClick={() => setPage(page + 1)}
            >
                Next
            </button>
        </nav>
    );
};

Paginator.propTypes = {
    page: PropTypes.number,
    setPage: PropTypes.func.isRequired,
    maxPage: PropTypes.number,
}

Paginator.defaultProps = {
    page: 0,
    maxPage: 0,
}

export default Paginator;
