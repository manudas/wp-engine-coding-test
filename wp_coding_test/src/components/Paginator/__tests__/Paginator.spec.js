import {
    render,
} from '@testing-library/react';

import Paginator from '..';

describe('Test suite for <Paginator>', () => {

    it('should render', () => {
        const { container } = render(
            <Paginator
                setPage={Function.prototype}
            />
        );
        expect(container).toMatchSnapshot();
    });

    it("previous should be disabled on first page and next shouldn't if more than 1 page", () => {
        const { getByText } = render(
            <Paginator
                page={0}
                maxPage={1}
                setPage={Function.prototype}
            />
        );

        const previousButton = getByText(/previous/i);
        const nextButton = getByText(/next/i);

        expect(previousButton).toHaveClass('disabled');
        expect(nextButton).not.toHaveClass('disabled');
    });


    it("previous shouldn't be disabled on second page and next should if last page", () => {
        const { getByText } = render(
            <Paginator
                page={1}
                maxPage={1}
                setPage={Function.prototype}
            />
        );

        const previousButton = getByText(/previous/i);
        const nextButton = getByText(/next/i);

        expect(previousButton).not.toHaveClass('disabled');
        expect(nextButton).toHaveClass('disabled');
    });
});