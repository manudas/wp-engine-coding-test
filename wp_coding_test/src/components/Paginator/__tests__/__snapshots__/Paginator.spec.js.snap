// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for <Paginator> should render 1`] = `
<div>
  <nav>
    <button
      class="paginator__button disabled"
    >
      Previous
    </button>
    <span
      class="paginator__page"
    >
      Page 
      1
       out of 
      1
    </span>
    <button
      class="paginator__button disabled"
    >
      Next
    </button>
  </nav>
</div>
`;
